hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/S3InMemoryCache.java

```diff
@@ -49,10 +51,38 @@ public void invalidate(String id) {
   }
 
   /**
-   * Clears the cache, removing all entries.
+   * Clears the cache by removing entries that correspond to transactions
+   * flushed by the doubleBuffer.
+   *
+   * @param flushedTransactionIds A list of transaction IDs that have been
+   *                              flushed and should be used to identify and
+   *                              remove corresponding cache entries.
    */
-  public void clearCache() {
-    cache.invalidateAll();
+  @Override
+  public void clearCache(List<Long> flushedTransactionIds) {
+    // Create a map to store transactionLogIndex-to-cacheKey mappings
+    Map<Long, Set<String>> transactionIdToCacheKeys = new HashMap<>();
```

Comment:
Can one transactionLogIdx be mapped to multiple cache keys ? I don't think so.

Reply:
Yes you are right every Ratis transaction will have a different index. I'll change that!
Thanks @sadanand48